mimic/rest/auth_api.py

```diff
@@ -151,17 +153,45 @@ def get_impersonation_token(self, request):
         except ValueError:
             request.setResponseCode(400)
             return json.dumps(invalid_resource("Invalid JSON request body"))
-
+        impersonator_token = request.getHeader("x-auth-token")
         expires_in = content['RAX-AUTH:impersonation']['expire-in-seconds']
         username = content['RAX-AUTH:impersonation']['user']['username']
-
+        impersonated_token = 'impersonated_token_' + text_type(uuid4())
         session = self.core.sessions.session_for_impersonation(username,
-                                                               expires_in)
+                                                               expires_in,
+                                                               impersonator_token,
+                                                               impersonated_token)
+
         return json.dumps({"access": {
-            "token": {"id": session.token,
+            "token": {"id": impersonated_token,
                       "expires": format_timestamp(session.expires)}
         }})
 
+    @app.route('/v2.0/tokens/<string:token_id>', methods=['GET'])
```

Comment:
I think we might want to consider starting to include links in each docstring to the openstack API being emulated.  In this case, http://developer.openstack.org/api-ref-identity-v2.html#admin-tokens